ui/src/features/annotator/tools/segment-anything-tool/segment-anything-tool.component.tsx

```diff
@@ -0,0 +1,225 @@
+
+import { useZoom } from '../../../../components/zoom/zoom';
+import { AnnotationShape } from '../../annotation-shape';
+import { isRightButton } from '../../annotation/utils';
+import { Annotation, Point, RegionOfInterest, Shape } from '../../interfaces';
+import { MaskAnnotations } from '../../mask-annotations';
+import { getRelativePoint, removeOffLimitPoints } from '../../shapes/utils';
+import { InteractiveSegmentationPoint } from './interactive-segmentation-point.component';
+import { useSegmentAnything } from './segment-anything-state-provider.component';
+import { InteractiveAnnotationPoint } from './segment-anything.interface';
+import { SvgToolCanvas } from './svg-tool-canvas.component';
+import { useSingleStackFn } from './use-single-stack-fn.hook';
+import { useThrottledCallback } from './use-throttled-callback.hook';
+
+import classes from './segment-anything.module.scss';
+
+const isPositivePoint = (point: Point, shapes: Shape[], isRightClick: boolean, rightClickMode: boolean) => {
+    if (rightClickMode) {
+        return !isRightClick;
+    }
+
+    return !shapes.some((shape) => isPointInShape(shape, point));
+};
+
+// Whenever the user moves their mouse over the canvas  we compute a preview of
+// SAM being applied to the user's mouse position.
+// The decoding step of SAM takes on average 100ms with 150-250ms being a high
+// exception. We throttle the mouse update based on this so that we don't overload
+// the user's cpu with too many decoding requests
+const THROTTLE_TIME = 150;
+
+const selectedMediaItem = {
+    identifier: 'id',
+    image: new ImageData(100, 100),
+};
+
+const roi: RegionOfInterest = {
+    x: 0,
+    y: 0,
+    width: 100,
+    height: 100,
+};
+
+const toolSettings = {
+    interactiveMode: false,
+    rightClickMode: false,
+};
+
+const SELECT_ANNOTATION_STYLES = {
+    fillOpacity: 0.3,
+    fill: 'var(--energy-blue-shade)',
+    stroke: 'var(--energy-blue-shade)',
+    strokeWidth: 'calc(2px / var(--zoom-level))',
+};
+
+export const SegmentAnythingTool = () => {
+    const zoom = useZoom();
+
+    const clampPoint = clampPointBetweenImage(selectedMediaItem.image);
+
+    const ref = useRef<SVGRectElement>(null);
+
+    const { result, points, addPoint } = useSegmentAnything();
+    const [mousePosition, setMousePosition] = useState<InteractiveAnnotationPoint>();
+
+    const [previewShapes, setPreviewShapes] = useState<Shape[]>([]);
+    const { decodingQueryFn } = useSegmentAnything();
+    const throttledDecodingQueryFn = useSingleStackFn(decodingQueryFn);
+
+    const throttleSetMousePosition = useThrottledCallback((point: InteractiveAnnotationPoint) => {
+        setMousePosition(point);
+    }, THROTTLE_TIME);
+
+    useEffect(() => {
+        if (mousePosition === undefined) {
+            return;
+        }
+
+        throttledDecodingQueryFn([mousePosition])
+            .then((shapes) => {
+                setPreviewShapes(shapes.map((shape) => removeOffLimitPoints(shape, roi)));
```

Comment:
There's quite some extra complexity from this PR due to `removeOffLimitPoints` (I think, since it uses a bunch of clipper stuff).
However since we don't expect to have task chain in Geti Tune the `roi` will be the full size of the image and the internals of SAM only return points within the image. So I think we can simply replace this by `setPreviewShapes(shapes)`.